contents/security/passwordStorage.md

```diff
@@ -0,0 +1,150 @@
+
+Additionally, even a small change in the message will result in a completely different <trigger for="pop:ciphertext">ciphertext</trigger>. This means that even very similar passwords that vary only by capitalization or a letter would still result in different ciphertext, so attackers can't compare the hash to try and figure out which users have similar passwords.
+
+<popover id="pop:ciphertext" title="_Ciphertext_ refers to encrypted information" placement="top">
+  <div slot="content">
+It is a cryptography term generally referring to data after encrypting it.
+  </div>
+</popover>
+
+For example, the following python code encrypts and decrypts the message "EncryptionIsFun!" with <trigger for="pop:aes">AES</trigger>:
+```python
+from Crypto.Cipher import AES
+
+AESCipher = AES.new("Here's a AES Key", AES.MODE_CBC, "Here's a AES IV.")
+message = "EncryptionIsFun!"
+ciphertext = AESCipher.encrypt(message)
+
+AESCipher2 = AES.new("Here's a AES Key", AES.MODE_CBC, "Here's a AES IV.")
+plaintext = AESCipher2.decrypt(ciphertext)
+
+print(plaintext))
+```
+
+<popover id="pop:aes" title="_AES_ stands for Advanced Encryption Standard" placement="top">
+  <div slot="content">
+It is one of many possible encryption algorithms.
+  </div>
+</popover>
+
+Encryption might seem like a good idea because it transfoms the data into essentially random text, which prevents all of the problems with storing the data directly in plaintext. However, because encryption is <tooltip content="If a function is reversible, and it converts from x to y, then it can also convert from y back to x"> reversible</tooltip>, it is always possible to regain the original password from the <trigger for="pop:ciphertext">ciphertext</trigger>. Since the data is encrypted, the <trigger for="pop:encrypt-key">encryption key</trigger> must also be stored somewhere. This means that if someone manages to hack into the application and read the encrypted passwords, it is also likely that they will be able to read the encryption key. With the encryption key, they will be able to decrypt all the passwords and read them anyway. This makes encryption unsuitable for password storage.
+
+<popover id="pop:encrypt-key" title="All encryption algorithms require an _encryption key_" placement="top">
+  <div slot="content">
+The key is usually randomly generated text that is used to encrypt the original data. To then decrypt the data, the same key (and algorithm) must be used.
+  </div>
+</popover>
+
+## What is hashing?
+
+Hashing is a <trigger for="pop:oneway">one-way</trigger> function that transforms a set of data into another set of data. Unlike encryption, hashing works by re-iterating on the input multiple times, each time using values that are dependent on the previous state, then destroying the previous state. This makes it impossible to reverse as a computer would have to test all possible states on each iteration, which compounds into several billion possible states. 
```

Comment:
I'm doubtful if this is definition of hashing in general. If feels like you are taking a specific use of hashing and giving it as the general definition.